### PR TITLE
Several improvements

### PR DESCRIPTION
**Pull request by mchv**
* Improve filtering by not writing to a temporary file
* Enable tests to run on continuous integration
* Handle correctly the `platform` parameter for ophan

### CODE REVIEW

**Comment by mchv** [comment]
This need to be deployed before [alexa skills](https://github.com/guardian/alexa-skills-lambda/pull/85) to be deployed on `PROD`

File: src/main/scala/com/gu/contentapi/Lambda.scala [hunk 1]
```diff
@@ -21,15 +22,16 @@ class Lambda extends RequestHandler[S3Event, Unit] with StrictLogging {
       .toList
 
     logObjects foreach { logObj =>
-      WriteToFile.fromLogObject(logObj, s"/tmp/${logObj.getKey}")
 
-      val allFastlyLogs = Source.fromFile(s"/tmp/${logObj.getKey}")("ISO-8859-1").getLines().toSeq flatMap { line =>
+      val allFastlyLogs: Seq[FastlyLog] = Source.fromBytes(IOUtils.toByteArray(logObj.getObjectContent))("ISO-8859-1").getLines().toSeq flatMap { line =>
```
Comment (oilnam) [inline src/main/scala/com/gu/contentapi/Lambda.scala:26]: good call 👍

File: src/test/scala/com/gu/contentapi/LambdaSpec.scala [hunk 2]
```diff
@@ -80,74 +80,90 @@ class LambdaSpec extends FlatSpec with Matchers with OptionValues {
...
-  //
-  //    logsGrouped.get("206").value should be(71)
-  //    logsGrouped.get("200").value should be(12)
-  //    logsGrouped.get("304").value should be(3)
-  //
-  //    val events = logs flatMap (Event(_))
-  //
-  //    events.length should be(15)
-  //  }
+  //TODO: the following tests should be treated as integration tests rather than being commented out
```
Comment (oilnam) [inline src/test/scala/com/gu/contentapi/LambdaSpec.scala:83]: we can remove this since you fixed it :)

**Comment by oilnam** [comment]
Looks great, the only thing I slightly disagree with is that you moved the filtering logic to the `FastlyLog` object. 

This means that now you have to explicitly apply the filter when transforming a `FastlyLog` to an `Event`, as you do in 

```scala
val events = logs.filter(FastlyLog.onlyDownloads).flatMap(Event(_))
```

whereas the way I did it (with the filtering inside the `Event.apply`) was transparent, and you could just do

```scala
val events = logs flatMap Event (_)
```

which is prettier. Also, the way you did it means that we have to remember to apply the filter every time.

What do you think?

**Comment by mchv** [comment]
@oilnam the filtering should certainly not be done in the `apply` method of event, which is not link to how you want to apply the filtering. I will come to explain

**Comment by mchv** [comment]
> whereas the way I did it (with the filtering inside the Event.apply) was transparent, and you could just do

 in other words it should not be transparent

**Comment by oilnam** [comment]
👍

**Pull request merged**